tests/integration/data_sources_and_expectations/test_misconfigured_expectations.py

```diff
@@ -0,0 +1,192 @@
+import datetime as dt
+
+import pandas as pd
+import pytest
+
+import great_expectations.expectations as gxe
+from great_expectations.datasource.fluent.interfaces import Batch
+from tests.integration.conftest import parameterize_batch_for_data_sources
+from tests.integration.data_sources_and_expectations.test_canonical_expectations import (
+    ALL_DATA_SOURCES,
+)
+from tests.integration.test_utils.data_source_config.base import DataSourceTestConfig
+
+PANDAS_DATA_SOURCES: list[DataSourceTestConfig] = [
+    ds for ds in ALL_DATA_SOURCES if ds.label.startswith("pandas")
+]
+SPARK_DATA_SOURCES: list[DataSourceTestConfig] = [
+    ds for ds in ALL_DATA_SOURCES if ds.label.startswith("spark")
+]
+SQL_DATA_SOURCES: list[DataSourceTestConfig] = [
+    ds for ds in ALL_DATA_SOURCES if ds not in PANDAS_DATA_SOURCES and ds not in SPARK_DATA_SOURCES
+]
+
+
+@pytest.mark.unit
+def test_parameterization():
+    # Ensure that all data sources are covered
+    assert len(PANDAS_DATA_SOURCES) + len(SPARK_DATA_SOURCES) + len(SQL_DATA_SOURCES) == len(
+        ALL_DATA_SOURCES
+    )
+
+
+def assert_misconfiguration(
+    batch_for_datasource: Batch, expectation: gxe.Expectation, exception_message: str
+) -> None:
+    result = batch_for_datasource.validate(expectation)
+    assert not result.success
+    assert exception_message in str(result.exception_info)
+
+
+class TestNumericExpectationAgainstStrDataMisconfiguration:
+    # Currently bugs with the following (not raising misconfiguration errors at all):
+    #  - sqlite
+    #  - databricks
+    #  - mysql
+    #  - spark-filesystem-csv
+
+    _DATA = pd.DataFrame({"a": ["b", "c"]})
+    _EXPECTATION = gxe.ExpectColumnStdevToBeBetween(
+        column="a",
+        min_value=0,
+        max_value=1,
+        strict_min=True,
+        strict_max=True,
+    )
+
+    @parameterize_batch_for_data_sources(
+        data_source_configs=PANDAS_DATA_SOURCES,
+        data=_DATA,
+    )
+    def test_pandas(self, batch_for_datasource) -> None:
+        assert_misconfiguration(
+            batch_for_datasource=batch_for_datasource,
+            expectation=self._EXPECTATION,
+            exception_message="could not convert string to float",
+        )
+
+    @parameterize_batch_for_data_sources(
+        data_source_configs=list(filter(lambda d: d.label == "big-query", SQL_DATA_SOURCES)),
```

Comment:
Probably simpler to just do `data_source_configs=[BigQueryTestConfig()]` that mess around with filtering and stuff.